packages/shopify-api/docs/reference/auth/tokenExchange.md

```diff
@@ -0,0 +1,77 @@
+# shopify.auth.tokenExchange
+
+Begins the OAuth process by exchanging the current user's [session token](https://shopify.dev/docs/apps/auth/session-tokens) for an
+[access token](https://shopify.dev/docs/apps/auth/access-token-types/online.md) to make authenticated Shopify API queries.
+
+Learn more:
+- [Token Exchange](../../guides/oauth.md#token-exchange)
+
+## Examples
+
+### Node.js
+```ts
+app.get('/auth', async (req, res) => {
+  const shop = shopify.utils.sanitizeShop(req.query.shop, true)
+  const headerSessionToken = getSessionTokenHeader(request);
+  const searchParamSessionToken = getSessionTokenFromUrlParam(request);
+  const sessionToken = (headerSessionToken || searchParamSessionToken)!;
+
+  await shopify.auth.tokenExchange({
+    sessionToken,
```

Comment:
I totally failed to notice this before so it's on me, but couldn't we accept the request (as a `rawRequest` value) here, and pull the header / search param out ourselves?

We'd also be able to pull the shop, so the only thing they'd need to change here is the token type.

Reply:
We don't necessarily want to do that because we don't know at what point the app is going to perform token exchange. They may have already parsed the request and gotten the headers. IMO, it's better to be specific in these params.

Reply:
It should be dirt cheap to get these values from the request, so I don't think the repetition would be a big deal. Not the most important thing but it creates a different API than the rest of the package (where we basically use the request for everything else).

I don't feel super strongly about this, but I'd still lean toward taking in the request.

Reply:
Yeah, we can revisit this while the flag is marked as unstable so there's no rush.